x-pack/plugins/security_solution/public/common/components/drag_and_drop/draggable_wrapper.tsx

```diff
@@ -115,26 +119,44 @@ export const getStyle = (
   };
 };
 
+const draggableContainsLinks = (draggableElement: HTMLDivElement | null) => {
+  const links = draggableElement?.querySelectorAll('.euiLink') ?? [];
+  return links.length > 0;
+};
+
 const DraggableWrapperComponent: React.FC<Props> = ({
   dataProvider,
   onFilterAdded,
   render,
   timelineId,
   truncate,
 }) => {
+  const keyboardHandlerRef = useRef<HTMLDivElement | null>(null);
   const draggableRef = useRef<HTMLDivElement | null>(null);
   const [closePopOverTrigger, setClosePopOverTrigger] = useState(false);
   const [showTopN, setShowTopN] = useState<boolean>(false);
   const [goGetTimelineId, setGoGetTimelineId] = useState(false);
   const timelineIdFind = useGetTimelineId(draggableRef, goGetTimelineId);
   const [providerRegistered, setProviderRegistered] = useState(false);
   const isDisabled = dataProvider.id.includes(`-${ROW_RENDERER_BROWSER_EXAMPLE_TIMELINE_ID}-`);
+  const [hoverActionsOwnFocus, setHoverActionsOwnFocus] = useState<boolean>(false);
   const dispatch = useDispatch();
 
-  const handleClosePopOverTrigger = useCallback(
-    () => setClosePopOverTrigger((prevClosePopOverTrigger) => !prevClosePopOverTrigger),
-    []
-  );
+  const handleClosePopOverTrigger = useCallback(() => {
+    setClosePopOverTrigger((prevClosePopOverTrigger) => !prevClosePopOverTrigger);
+    setHoverActionsOwnFocus((prevHoverActionsOwnFocus) => {
+      if (prevHoverActionsOwnFocus) {
+        setTimeout(() => {
+          keyboardHandlerRef.current?.focus();
+        }, 0);
+      }
+      return false; // always give up ownership
+    });
+
+    setTimeout(() => {
+      setHoverActionsOwnFocus(false);
+    }, 0); // invoked on the next tick, because we want to restore focus first
+  }, [keyboardHandlerRef, setClosePopOverTrigger, setHoverActionsOwnFocus]);
```

Comment:
nit -> you do not need the state updater as dependency since it is directly returned from useState and doesn't change during the course of your component Lifecycle you can avoid passing it as a dependency.

As [docs states](https://reactjs.org/docs/hooks-reference.html#usestate):
React guarantees that setState function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the useEffect or useCallback dependency list.

https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies